src/main/java/com/redhat/devtools/intellij/tektoncd/tree/TektonRootNode.java

```diff
@@ -25,6 +25,7 @@ public TektonRootNode(Project project) {
   }
 
   public CompletableFuture<Tkn> initializeTkn() {
+    TknCliFactory.getInstance().setTkn(project);
```

Comment:
I don't get why do you need to insert your code as calling getTkn will cause setTkn to be called

Reply:
setTkn (in getTkn) is only called if future is null. If you change your config once it has been already initialized, future will be different from null and setTkn will never be called. This raise a possible issue.

An example.... when you start the plugin, tkn client is initialized by using the current config (suppose admin/cluster), if you at a certain time switch user to `developer` (current config becomes developer/cluster), the client is never updated and it will remain the same initialized with the old config which is wrong.

Reply:
I think it would be more understandable to do:
1) add a reset method that would be called on the load method of the root node and that method would reset the future to null
2) keep initializeTkn as it was

WDYT ?

Reply:
Ok for me